packages/mui-material/src/Select/SelectInput.js

```diff
@@ -325,6 +325,8 @@ const SelectInput = React.forwardRef(function SelectInput(props, ref) {
   if (isFilled({ value }) || displayEmpty) {
     if (renderValue) {
       display = renderValue(value);
+    } else if (Array.isArray(value) && value.length === 0) {
```

Comment:
this might be conceptional cleaning to do below inside the if (computeDisplay) block where the problematic reduce is. 

at this point in the code, we are doing a clean check of 'if the user has a custom render, use it, if not use ours'.

really 'ours' should be robust enough to not break, so I think doing this check where the reduce is keeps the logic to 'compute the display' all in one place (where 'compute the display' means empty string for the empty array case)

Reply:
I agree. I changed it